reactor-test/src/main/java/reactor/test/util/RaceTestUtils.java

```diff
@@ -99,77 +101,79 @@
 		}
 	}
 
+	public static void race(final Runnable r1, final Runnable r2) {
+		race(new Runnable[]{r1, r2});
+	}
+
 	/**
-	 * Synchronizes the execution of two {@link Runnable} as much as possible
-	 * to test race conditions. The method blocks until both have run to completion.
-	 * @param r1 the first runnable
-	 * @param r2 the second runnable
+	 * Synchronizes the execution of several {@link Runnable}s as much as possible
+	 * to test race conditions. The method blocks until all have run to completion.
+	 * @param rs the runnables to execute
 	 */
-	public static void race(final Runnable r1, final Runnable r2) {
-		race(r1, r2, Schedulers.single());
+	public static void race(final Runnable... rs) {
+		race(Schedulers.parallel(), rs);
 	}
 
 	/**
 	 * Synchronizes the execution of two {@link Runnable} as much as possible
 	 * to test race conditions. The method blocks until both have run to completion.
+	 * @param s the {@link Scheduler} on which to execute the runnables
 	 * @param r1 the first runnable
 	 * @param r2 the second runnable
-	 * @param s the {@link Scheduler} on which to execute the runnables
+	 * @deprecated use {@link #race(Scheduler, Runnable...)}
 	 */
+	@Deprecated
 	public static void race(final Runnable r1, final Runnable r2, Scheduler s) {
-		final AtomicInteger count = new AtomicInteger(2);
-		final CountDownLatch cdl = new CountDownLatch(2);
-
-		final Throwable[] errors = { null, null };
+		race(s, r1, r2);
+	}
 
-		s.schedule(() -> {
-			if (count.decrementAndGet() != 0) {
-				while (count.get() != 0) { }
-			}
+	/**
+	 * Synchronizes the execution of several {@link Runnable}s as much as possible
+	 * to test race conditions. The method blocks until both have run to completion.
+	 * @param s the {@link Scheduler} on which to execute the runnables
+	 * @param rs the runnables to execute
+	 */
+	public static void race(Scheduler s, final Runnable... rs) {
+		final AtomicInteger count = new AtomicInteger(rs.length);
+		final CountDownLatch cdl = new CountDownLatch(rs.length);
+		final Throwable[] errors = new Throwable[rs.length];
+		for (int i = 0; i < rs.length; i++) {
+			final int index = i;
+			s.schedule(() -> {
+				if (count.decrementAndGet() != 0) {
+					while (count.get() != 0) { }
+				}
 
-			try {
 				try {
-					r1.run();
-				} catch (Throwable ex) {
-					errors[0] = ex;
+					try {
+						rs[index].run();
+					} catch (Throwable ex) {
+						errors[index] = ex;
+					}
+				} finally {
+					cdl.countDown();
 				}
-			} finally {
-				cdl.countDown();
-			}
-		});
-
-		if (count.decrementAndGet() != 0) {
-			while (count.get() != 0) { }
-		}
-
-		try {
-			try {
-				r2.run();
-			} catch (Throwable ex) {
-				errors[1] = ex;
-			}
-		} finally {
-			cdl.countDown();
+			});
 		}
 
 		try {
-			if (!cdl.await(5, TimeUnit.SECONDS)) {
+			if (!cdl.await(50, TimeUnit.SECONDS)) {
```

Comment:
is this bump of timeout on purpose, or a leftover of debugging?

Reply:
yup, shouldn't be needed at all.